File: easy/LongestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
       String commonprefix=strs[0];
       for(int i=1;i<strs.length;i++){
        String c=strs[i];
        StringBuilder temp=new StringBuilder();
        for(int j=0;j<Math.min(c.length(),commonprefix.length());j++){
            char c1=strs[i].charAt(j);
            char c2=commonprefix.charAt(j);
            if(c1!=c2)
              break;
              temp.append(c1);
        }
        commonprefix=temp.toString();
       }
       return commonprefix;
       //tc: o(n*m)
    }
}

File: easy/LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string commonprefix = strs[0];
        for (int i = 1; i < strs.size(); i++) {
            string temp = "";
            for (int j = 0; j < min(strs[i].size(), commonprefix.size()); j++) {
                if (strs[i][j] != commonprefix[j]) break;
                temp.push_back(strs[i][j]);
            }
            commonprefix = temp;
        }
        return commonprefix;
    }
};